src/helm/proxy/clients/huggingface_model_registry.py

```diff
@@ -71,36 +82,73 @@ def from_string(raw: str) -> "HuggingFaceModelConfig":
             namespace=match.group("namespace"), model_name=model_name, revision=match.group("revision")
         )
 
+    @staticmethod
+    def from_path(path: str) -> "HuggingFaceModelConfig":
+        """Generates a HuggingFaceModelConfig from a (relative or absolute) path to a local HuggingFace model."""
+        pattern = r"((?P<containing_dir>.+?)/)?(?P<model_name>[^/@]+)(@(?P<revision>[^/@]+))?/?"
+        match = re.fullmatch(pattern, path)
+        if not match:
+            raise ValueError(f"Could not parse model path: '{path}'; Expected format: /path/to/model_name[@revision]")
+        model_name = match.group("model_name")
+        assert model_name
+        return HuggingFaceModelConfig(
+            namespace="huggingface",
+            model_name=model_name,
+            revision=match.group("revision"),
+            path=path
+        )
 
-_huggingface_model_registry: Dict[str, HuggingFaceModelConfig] = {}
-
-
-def register_huggingface_model_config(model_name: str) -> HuggingFaceModelConfig:
-    """Register a AutoModelForCausalLM model from Hugging Face Model Hub for later use.
-
-    model_name format: namespace/model_name[@revision]"""
-    config = HuggingFaceModelConfig.from_string(model_name)
+_huggingface_model_registry: Dict[str, HuggingFaceModelConfig] = {
+    "huggingface/llama-7b": HuggingFaceModelConfig(
```

Comment:
Do we need to put an entry for each model?  There might be a lot of open models...  Also, wonder if this could be unified with `models.py`...otherwise we have three places we have to define each model (schema.yaml, models.py, and this file)....

Reply:
yep I agree the duplication isn't great

Reply:
fixed